os/fs/driver/mtd/smart.c

```diff
@@ -3617,19 +3615,20 @@ static int smart_garbagecollect(FAR struct smart_struct_s *dev)
 				count = smart_get_count(dev, dev->releasecount, x);
 				if (count > releasemax) {
 					releasemax = count;
-					collectblock = x;
-				}
 #else
 				if (dev->releasecount[x] > releasemax) {
 					releasemax = dev->releasecount[x];
+#endif
```

Comment:
I prefer previous style it is more good readability
```
#ifdef A
    if (true) {
      do_something;
   }
#else
   if (true) {
     do_something;
   }
#endif

Reply:
The only difference between this suggestion and the old version is that with this suggestion we avoid duplicating 1 line of code.


```
#ifdef A
    if () {
        do something first based on A;
        then do something more
    }
#else
    if () {
        do something first based on B;
        then do something more;
    }
#endif
```

Has been changed to : -
 
```
#ifdef A
    if () {
        do something first base on A;
    }
#else
    if () {
        do something first based on B;
    }
#endif
    then do something more;
```